src/contexts/UploadContext.tsx

```diff
@@ -0,0 +1,29 @@
+'use client'
+import { createContext, useContext, useState, PropsWithChildren } from 'react'
+
+interface UploadContextProps {
+  triggerUpload: boolean
+  setTriggerUpload: React.Dispatch<React.SetStateAction<boolean>>
+  uploadSuccess: boolean
+  setUploadSuccess: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+export const UploadProvider = ({ children }: PropsWithChildren) => {
+  const [triggerUpload, setTriggerUpload] = useState(false)
+  const [uploadSuccess, setUploadSuccess] = useState(false)
+
```

Comment:
**nitpick:** Let's add uploading logic to this provider here, so you just extract upload method and use it anywhere you want